virtualmachine/vsphere/vm.go

```diff
@@ -429,6 +429,72 @@ func (vm *VM) GetName() string {
 	return vm.Name
 }
 
+func (vm *VM) AddDisk() (err error) {
+	if err := SetupSession(vm); err != nil {
+		return fmt.Errorf("Error setting up vSphere session: %s", err)
+	}
+
+	// Cancel the sdk context
+	defer vm.cancel()
+
+	// Get a reference to the datacenter with host and vm folders populated
+	dcMo, err := GetDatacenter(vm)
+	if err != nil {
+		return fmt.Errorf("Failed to retrieve datacenter: %s", err)
+	}
+
+	vmMo, err := findVM(vm, dcMo, vm.Name)
+	if err != nil {
+		return fmt.Errorf("VM not found", vm.Name, err)
+	}
+
+	n := util.Random(1, len(vm.Datastores))
+	vm.datastore = vm.Datastores[n-1]
+	if err := reconfigureVM(vm, vmMo); err != nil {
+		return fmt.Errorf("Reconfigure failed : ", err)
+	}
+	return nil
+}
+
+func (vm *VM) RemoveDisk() (err error) {
+	if err := SetupSession(vm); err != nil {
+		return fmt.Errorf("Error setting up vSphere session: %s", err)
+	}
+
+	// Cancel the sdk context
+	defer vm.cancel()
+
+	// Get a reference to the datacenter with host and vm folders populated
+	dcMo, err := GetDatacenter(vm)
+	if err != nil {
+		return fmt.Errorf("Failed to retrieve datacenter: %s", err)
+	}
+
+	vmMo, err := findVM(vm, dcMo, vm.Name)
+	if err != nil {
+		return fmt.Errorf("VM not found", vm.Name, err)
+	}
+	var deviceMo *types.VirtualDisk
+	spec := new(types.VirtualMachineConfigSpec)
+	for _, d := range vmMo.Config.Hardware.Device {
+		switch device := d.(type) {
+		case *types.VirtualDisk:
+			deviceMo = device
```

Comment:
It need get the device name and compare that and remove the matched device only.